store/tikv/gcworker/gc_worker.go

```diff
@@ -515,63 +517,85 @@ func doGC(ctx goctx.Context, store tikv.Storage, safePoint uint64, identifier st
 	// Sleep to wait for all other tidb instances update their safepoint cache.
 	time.Sleep(gcSafePointCacheInterval)
 
-	req := &tikvrpc.Request{
-		Type: tikvrpc.CmdGC,
-		GC: &kvrpcpb.GCRequest{
-			SafePoint: safePoint,
-		},
-	}
-	bo := tikv.NewBackoffer(tikv.GcMaxBackoff, goctx.Background())
-
 	log.Infof("[gc worker] %s start gc, safePoint: %v.", identifier, safePoint)
 	startTime := time.Now()
 	regions := 0
+	errRegions := 0
 
+	ticker := time.NewTicker(gcJobLogTickInterval)
+	defer ticker.Stop()
+
+	bo := tikv.NewBackoffer(tikv.GcLocalRegionMaxBackoff, goctx.Background())
 	var key []byte
 	for {
 		select {
 		case <-ctx.Done():
 			return errors.New("[gc worker] gc job canceled")
+		case <-ticker.C:
+			log.Infof("[gc worker] %s gc in process, safePoint: %v, regions: %v, ignored regions: %v, cost time: %s",
+				identifier, safePoint, regions, errRegions, time.Since(startTime))
 		default:
 		}
 
 		loc, err := store.GetRegionCache().LocateKey(bo, key)
 		if err != nil {
 			return errors.Trace(err)
 		}
-		resp, err := store.SendReq(bo, req, loc.Region, tikv.ReadTimeoutLong)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		regionErr, err := resp.GetRegionError()
+
+		var regionErr *errorpb.Error
+		regionErr, err = doGCForOneRegion(store, safePoint, loc)
+
 		if err != nil {
-			return errors.Trace(err)
-		}
-		if regionErr != nil {
-			err = bo.Backoff(tikv.BoRegionMiss, errors.New(regionErr.String()))
-			if err != nil {
-				return errors.Trace(err)
-			}
+			errRegions++
+		} else if regionErr != nil {
```

Comment:
if regionErr != nil {
 continue
}
if err != nil {
} else {
}

Reply:
I think it is better to check err first, because if `regionErr` and `err` both exist, `err` should have higher priority.